communication/mavlink_communication.cpp

```diff
@@ -40,221 +40,105 @@
-#include "util/print_util.h"
+#include "util/print_util.hpp"
 #include "hal/common/time_keeper.hpp"
 
 //------------------------------------------------------------------------------
-// PRIVATE FUNCTIONS IMPLEMENTATION
-//------------------------------------------------------------------------------
-
-bool Mavlink_communication::send_message(send_msg_handler_t* msg_send)
-{
-    bool success = true;
-
-    Mavlink_communication::send_msg_function_t function = msg_send->function;
-    handling_telemetry_module_struct_t module_struct = msg_send->module_struct;
-
-    mavlink_message_t msg;
-    function(module_struct, msg_send->mavlink_stream, &msg);
-
-    success &= msg_send->mavlink_stream->send(&msg);
-
-    return success;
-}
-
-void Mavlink_communication::toggle_telemetry_stream(Scheduler* scheduler, uint32_t sysid, mavlink_message_t* msg)
-{
-    // Decode message
-    mavlink_request_data_stream_t request;
-    mavlink_msg_request_data_stream_decode(msg, &request);
-
-    if (((request.target_system == sysid) || (request.target_system == MAV_SYS_ID_ALL)))
-        //&&(request.target_component == 0))
-    {
-        if (request.req_stream_id == 255)
-        {
-            // send full list of streams
-            scheduler->run_all_tasks_now();
-        }
-        else
-        {
-            Scheduler_task* task = scheduler->get_task_by_id(request.req_stream_id);
-
-            if (task != NULL)
-            {
-                if (request.start_stop)
-                {
-                    task->set_run_mode(Scheduler_task::Scheduler_task::RUN_REGULAR);
-                }
-                else
-                {
-                    task->set_run_mode(Scheduler_task::Scheduler_task::RUN_NEVER);
-                }
-
-                if (request.req_message_rate > 0)
-                {
-                    task->change_period(Scheduler::TIMEBASE / (uint32_t)request.req_message_rate);
-                }
-            }
-            else
-            {
-                print_util_dbg_print("This stream ID is not registred and cannot be activated.\r\n");
-            }
-        } //end of else request.req_stream_id != 255
-    }
-}
-
-//------------------------------------------------------------------------------
 // PUBLIC FUNCTIONS IMPLEMENTATION
 //------------------------------------------------------------------------------
 
 Mavlink_communication::Mavlink_communication(Serial& serial, State& state, File& file_storage, const conf_t& config) :
-    scheduler_(config.scheduler_config),
     mavlink_stream_(serial, config.mavlink_stream_config),
-    message_handler_(mavlink_stream_, config.message_handler_config),
-    onboard_parameters_(scheduler_, file_storage, state, message_handler_, mavlink_stream_, config.onboard_parameters_config),
-    msg_sending_count_(0)
+    telemetry_(mavlink_stream_, config.telemetry_config),
+    handler_(mavlink_stream_, config.message_handler_config),
```

Comment:
I shortened to handler_ because it is shorter and also because the handler handles messages and commands.
renamed in https://github.com/lis-epfl/MAVRIC_Library/pull/326/commits/98910bc05cbe61feaf0f0bfff35093894e51533c